SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -110,47 +110,57 @@ Interest InterestFactory::CreateServerWorkerInterest(const UAbstractLBStrategy*
 
-	// and server worker entities
+	// And an interest in all server worker entities.
 	ServerQuery = Query();
 	ServerQuery.ResultComponentIds =
 		SchemaResultType{ SpatialConstants::SERVER_WORKER_COMPONENT_ID, SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
 	ServerQuery.Constraint.ComponentConstraint = SpatialConstants::SERVER_WORKER_COMPONENT_ID;
 	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
 
-	// Add a self query to ensure we see the well known entity tag.
-	Query AuthoritySelfQuery = {};
-	AuthoritySelfQuery.ResultComponentIds = { SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
-	AuthoritySelfQuery.Constraint.bSelfConstraint = true;
-	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::SERVER_WORKER_COMPONENT_ID, AuthoritySelfQuery);
-
-	// Query to know about all the actors tagged with a debug component
-	if (bDebug)
-	{
-		ServerQuery = Query();
-		ServerQuery.ResultComponentIds = SchemaResultType{ SpatialConstants::GDK_DEBUG_COMPONENT_ID };
-		ServerQuery.Constraint.ComponentConstraint = SpatialConstants::GDK_DEBUG_COMPONENT_ID;
-		AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
-	}
+	// Ensure server worker receives core GDK snapshot entities.
+	ServerQuery = Query();
+	ServerQuery.ResultComponentIds = ServerNonAuthInterestResultType;
+	ServerQuery.Constraint = CreateGDKSnapshotEntitiesConstraint();
+	AddComponentQueryPairToInterestComponent(ServerInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, ServerQuery);
 
 	return ServerInterest;
 }
 
-Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker) const
+Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker, bool bDebug) const
 {
 	// Add load balancing query
 	Interest PartitionInterest{};
+	Query PartitionQuery{};
+
 	AddLoadBalancingInterestQuery(LBStrategy, VirtualWorker, PartitionInterest);
+
+	// Ensure server worker receives AlwaysRelevant entities.
+	PartitionQuery = Query();
+	PartitionQuery.ResultComponentIds = ServerNonAuthInterestResultType;
+	PartitionQuery.Constraint = CreateAlwaysRelevantConstraint();
+	AddComponentQueryPairToInterestComponent(PartitionInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, PartitionQuery);
+
+	// Add a self query for completeness
+	PartitionQuery = Query();
+	PartitionQuery.ResultComponentIds = { SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
+	PartitionQuery.Constraint.bSelfConstraint = true;
+	AddComponentQueryPairToInterestComponent(PartitionInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, PartitionQuery);
+
+	// Query to know about all the actors tagged with a debug component
+	if (bDebug)
```

Comment:
I think this one should also be on the server worker entity- I would expect only one of this and the always relevant query to be made from each worker